ansible/roles/preflights/tasks/main.yaml

```diff
@@ -0,0 +1,393 @@
+---
+  - name: check for supported OS
+    fail:
+      msg: "{{ configured_ansible_distribution }} {{ ansible_distribution_major_version }} is not supported. The supported OSes - CentOS 7/8, RHEL 7/8, Suse 15, SLES 15, Ubuntu 16 (xenial) 18 (bionic) or Debian 9 (stretch)"
+    failed_when: not (
+                 (configured_ansible_distribution == "CentOS" and ansible_distribution_major_version|int in [7, 8])
+                 or (configured_ansible_distribution == "RedHat" and ansible_distribution_major_version|int in [7, 8])
+                 or (configured_ansible_distribution == 'Ubuntu' and ((ansible_distribution_major_version|int == 16) or (ansible_distribution_major_version|int == 18)))
+                 or (configured_ansible_distribution == 'Debian' and ((ansible_distribution_major_version|int == 9) or (ansible_distribution_major_version|int == 10)))
+                 or (ansible_distribution == 'Suse' and ansible_distribution_major_version|int == 15)
+                 or (ansible_distribution == 'SLES' and ansible_distribution_major_version|int == 15)
+                 or (ansible_distribution == 'openSUSE Leap' and ansible_distribution_major_version|int == 15)
+                 )
```

Comment:
Flatcar should be here too.
Also please verify the rest of the list is correct.